### ISSUE TITLE
Channels request return empty after switching user

### ISSUE THREAD

**Issue by lukenowakcom** [GetStream/stream-chat-android#35]
## What did you do?
**1.** Initialised client
**2.** Set user to user `X`
**3.** Sent query to fetch user `X` channels
**4.** Query successfully returned array of channels for user `X`
**4.5** Disconnected client(first I didn't do that, then I tried with it)
**5.** Set user to user `Y`
**6.** Sent query to fetch user `Y` channels
**7.** Query successfully returned but with empty array of channels for user `Y`
**8.** Completely restarted the app(StreamChat got reinitialised)
**9.** Set user to user `Y`
**10.** Sent query to fetch user `Y` channels
**11.** Query successfully returned array of channels for user `Y`

## What did you expect to happen?
To receive user `Y` channels after he got successfully set on the client.

## What happened instead?
The array of channels returned empty, despite passing  an exact `cid` in the request.

## GetStream Environment
**GetStream Chat version:**  1.8.7 with later commits(master)
**Android version:**  9.0.0
**Android Studio:** 3.5
**Device:** OnePlus6

## Additional context
**1.** Despite disconnecting the client or setting user `Y`, channels belonging to user `X` still hang around in `activeChannels` in client object.
**2.** Channels query is exactly the same every time with the exception of `clientId`(which changes after disconnecting client before setting new user) and `cid`(different for both users)

**Comment by tschellenbach** [comment]
that's an interesting bug. @tbarbugli thoughts?

**Comment by tbarbugli** [comment]
the SDK does not support this nicely, @lukenowakcom how do your switch user code looks like atm?

**Comment by lukenowakcom** [comment]
```
StreamChat.init("apiKey", application)
 _chatClient - StreamChat.getInstance(application)
```
```
if (_chatClient.isConnected) _chatClient.disconnect()
```
```
"fetching user token from backend and calling initUser upon completion"
```
```
private fun initUser(token: String) {
        _chatClient.onSetUserCompleted(object : ClientConnectionCallback {

            override fun onSuccess(user: User?) {
                _handler?.post {
                    initChannel()
                }
            }

            override fun onError(errMsg: String?, errCode: Int) {
            }
        })

        val userData = HashMap<String, Any>().apply {
            put("name", userModel.name
                    ?: "Error")
        }
        _chatClient.setUser(User(userMode.chatId, userData), token)
    }
```
```
val channelCid = "${userModel.userGetStreamChannelType}:${userModel.userGetStreamChannelId}"
        val query = QueryChannelsRequest(Filters.eq("cid", channelCid), QuerySort())
        _chatClient.queryChannels(query, object : QueryChannelListCallback {

            override fun onSuccess(response: QueryChannelsResponse?) {
                _handler?.post {
                    val channel = response?.channels?.getOrNull(0)
                    with(channel) {
                        if (this != null) {
                            _chatChannel.value = this
                            _chatChannelStatus.value = Status.READY
                        } else {
                            _chatChannelStatus.value = Status.FAILED
                        }
                    }
                }
            }

            override fun onError(errMsg: String?, errCode: Int) {
            }
        })
```

And then when first user logs out and another logs in, its all the same order, only in this case `StreamChat.init("apiKey", application)` is not called again since it throws Exception anyway.
Nothing is triggering `onError` callbacks. Its only that for the second user `QueryChannelsResponse` does not contain any channels until the app is killed and restarted and `StreamChat.init()` is called again.

*last tried on **2.0.1**.

**Comment by tbarbugli** [comment]
#96 should fix this

**Comment by lukenowakcom** [comment]
Just tried master, unfortunately the issue is still there :/

**Comment by tbarbugli** [comment]
@lukenowakcom the switch user PR was merged after your last message, can you try now?

**Comment by lukenowakcom** [comment]
Okey... so it works when you call `chatClient.disconnect()` prior to setting a new user, otherwise it returns empty array again.
But if you do call `chatClient.disconnect()` then the `StreamChat.getOnlineStatus()` seems to be stuck at `CONNECTING` at least until I pause the activity/fragment the chat views are in.

I tracked the states in `Application` class since the `StreamChat.init()` call(see below)

It started as usual with `NOT_INITIALIZED` then I logged in and got `CONNECTED`(not even `CONNECTING` in-between). 
Then I logged out and got `FAILED`.
Then I logged in into second user and got `CONNECTING`.
Then I locked the screen and got `FAILED`
Then unlocked it ad got `CONNECTING` followed by `CONNECTED`
```
2019-10-07 15:43:50.349 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: NOT_INITIALIZED 
2019-10-07 15:45:03.486 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: CONNECTED
2019-10-07 15:45:08.465 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: FAILED
2019-10-07 15:45:25.925 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: CONNECTING
2019-10-07 15:46:30.364 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: FAILED
2019-10-07 15:46:33.528 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: CONNECTING
2019-10-07 15:46:34.687 20906-20906/io.ahoy.ahoy.rc E/AhoyApp: CONNECTED
```

It's worth noting that despite being stuck at `CONNECTING` setting user and querying for channels worked just fine
